.github/workflows/main.yml

```diff
@@ -1,6 +1,9 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches: [ main ]
```

Comment:
This way the workflow doesn't run for branches unless there is a pull request for it. Is there a reason why you disabled that?

Reply:
merged too quickly I guess :sweat_smile:

Reply:
🚨👮‍♂️🚨😄 

It's anyway a small adjustment, I'll take care of it.

Reply:
Sorry to break the flow for local branches. I intended to skip multiple executions when you start working inside the fork's PR. Before the change each commit triggers the workflows two times.

If you need the experiments with local branches I suggest adding some additional pattern like:
```yml
on:
  push:
    branches:
      - main
      - local-pattern-here-*
```

Reply:
```

  ┌────────────────────────────┐       ┌─────────────────────────────┐
  │   Push feature branch      ├──────►│   Create Pull Request       │
  └─────────────┬──────────────┘       └──────────────┬──────────────┘
                │                                     │
                │                                     │
                │                                     │
 ┌──────────────▼──────────────────┐  ┌───────────────▼────────────┐
 │   Push event triggers workflow  │  │ PR event triggers workflow │
 └─────────────────────────────────┘  └────────────────────────────┘

```
I mean just pipeline execution, the tests summary step will be executed in each pipeline separately.

Reply:
This is true only for branches of this repo but not for forks.

Check [this PR](https://github.com/assertj/assertj/pull/2794/checks) for example, `CI` and `Cross-Version` are running only for `pull-request` events, leading to a total of 12 checks.

However, if you check a [dependabot PR](https://github.com/assertj/assertj/pull/2787/checks), `CI` and `Cross-Version` are running for both `push` and `pull-request` events, leading to a total of 22 checks.

Reply:
Let's properly address this topic in #2810.